wgpu-native/src/instance.rs

```diff
@@ -235,6 +236,16 @@ pub fn adapter_create_device(adapter_id: AdapterId, _desc: &DeviceDescriptor) ->
     let (raw, queue_group) = adapter.open_with::<_, hal::General>(1, |_qf| true).unwrap();
     let mem_props = adapter.physical_device.memory_properties();
     let limits = adapter.physical_device.limits();
+
+    assert!(
+        DYNAMIC_BUFFER_OFFSET_ALIGN % limits.min_storage_buffer_offset_alignment == 0,
+        "Device storage buffer offset alignment not compatible with WGPU"
+    );
+    assert!(
+        DYNAMIC_BUFFER_OFFSET_ALIGN % limits.min_uniform_buffer_offset_alignment == 0,
+        "Device uniform buffer offset alignment not compatible with WGPU"
+    );
```

Comment:
We could also check these in `Device::new` instead :man_shrugging:

Reply:
We should do it even earlier - as soon as we select an adapter